src/mongo/db/dbcommands_generic.cpp

```diff
@@ -275,12 +275,18 @@ class ListCommandsCmd : public Command {
                      int,
                      string& errmsg,
                      BSONObjBuilder& result) {
+        // sort the commands before building the result BSON
+        std::map<string, Command*> sortedCommands;
+        for (CommandMap::const_iterator it = _commands->begin(); it != _commands->end(); ++it) {
+            sortedCommands.emplace(it->first, it->second);
+        }
```

Comment:
Thank you for the comments.

I cannot use std::map<CommandMap::key_type, CommandMap::value_type> here because CommandMap::value_type itself is of type pair<string, Command*>.

I am not entirely convinced how std::vector based implementation will be faster than the std::map one. Because if you go step by step,
std::map takes a total of N*log(N) time (for insertion) whereas std::vector based implementation has a total complexity of N + N*log(N) (Insertion + Sorting). 
If you have based your arguement on the size of the input or on how these STL containers are implemented, could you please point me to some literature explaining these things. I am very much interested in knowing more about it.

Finally, I agree with you on performing filtering in the first iteration itself. I will go ahead and make that change.